packages/react/src/components/Chat/ChatMessage.tsx

```diff
@@ -86,20 +102,35 @@ class ChatMessage extends UIComponent<ReactProps<ChatMessageProps>, ChatMessageS
   }
 
   public state = {
+    focused: false,
     isFromKeyboard: false,
   }
 
+  protected actionHandlers: AccessibilityActionHandlers = {
+    // prevents default FocusZone behavior, e.g., in ChatMessageBehavior, it prevents FocusZone from using arrow keys as navigation (only Tab key should work)
+    preventDefault: event => {
+      event.preventDefault()
+    },
+  }
+
   private handleFocus = (e: React.SyntheticEvent) => {
+    this.changeFocusState(true)
     this.setState({ isFromKeyboard: isFromKeyboard() })
 
     _.invoke(this.props, 'onFocus', e, this.props)
   }
 
-  protected actionHandlers: AccessibilityActionHandlers = {
-    // prevents default FocusZone behavior, e.g., in ChatMessageBehavior, it prevents FocusZone from using arrow keys as navigation (only Tab key should work)
-    preventDefault: event => {
-      event.preventDefault()
-    },
+  handleBlur = (e: React.FocusEvent) => {
+    const shouldPreserveFocusState = _.invoke(e, 'currentTarget.contains', e.relatedTarget)
```

Comment:
could you, please, suggest the case we would like to cover with that?

Reply:
It comes from the accessibility point: when we will click `Enter` key, the focus will be moved to another element.

### With

![chat-with1](https://user-images.githubusercontent.com/14183168/52276008-b9f24000-2959-11e9-8a0a-c4f354a501b9.gif)

### Without

![chat-without](https://user-images.githubusercontent.com/14183168/52276011-bced3080-2959-11e9-9be1-0f463dba356e.gif)

Reply:
then, frankly, I am a bit confused with `onBlur` event being raised unconditionally - so, for instance, even if `shouldPreserveFocusState` is `true` (and, as a consequence,  even `this.state.focused` remains to be true) - even in this situation we will trigger `blur` event for `ChatMessage`.

So, couple of things that I would like to solve here:
- ensure that I am not missing anything in this thought 
- and, in case if so, it seems that we need to resolve this emerged problem that is about implicit reliance of events subscriptions - we should strive for component's behavioural intents being explicit

Could you, please, to resolve this issue for now, just provide a comment that will describe the intent of these lines? Thank you!

Reply:
`this.state.focused` controls is focused the `ChatMessage` on any of its children, when we're navigating with keyboard focused elements changes and there is no way to use `:focus`:

![image](https://user-images.githubusercontent.com/14183168/52281752-6be43900-2967-11e9-8ebb-c958578e5e7b.png)

May be proper naming can make it better?

Reply:
yes, totally agree about naming - this should help. However, it seems that many things are involved here, so it is, probably, much better to address naming and all the related moments in a dedicated PR - what do you think?